src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+
+        The visible region of an `Object`, whose shape depends on
+        :prop:`viewAngles`. For more details, see `DefaultViewRegion`.
+        """
+        true_position = self.position.offsetRotated(self.orientation, toVector(self.cameraOffset))
+        return DefaultViewRegion(visibleDistance=self.visibleDistance, viewAngles=self.viewAngles,\
+            position=true_position, rotation=self.orientation)
+
+    @cache
+    def canSee(self, other, occludingObjects=list(), debug=False) -> bool:
+        """Whether or not this `Object` can see ``other``.
+
+        Args:
+            other: A `Point`, `OrientedPoint`, or `Object` to check
+              for visibility.
+            occludingObjects: A list of objects that can occlude visibility.
+        """
+        true_position = self.position.offsetRotated(self.orientation, toVector(self.cameraOffset))
+        return canSee(position=true_position, orientation=self.orientation, visibleDistance=self.visibleDistance, \
+            viewAngles=self.viewAngles, rayDensity=self.rayDensity, target=other, \
+            occludingObjects=occludingObjects, debug=debug)
+
+    @cached_property
+    def corners(self):
+        """A tuple containing the corners of this object's bounding box"""
+        hw, hl, hh = self.hw, self.hl, self.hh
+        # Note: 2D show method assumes cyclic order of first 4 vertices
+        return (
+            self.relativePosition(Vector(self.hw,  self.hl,  self.hh)),
+            self.relativePosition(Vector(-self.hw, self.hl,  self.hh)),
+            self.relativePosition(Vector(-self.hw, -self.hl, self.hh)),
+            self.relativePosition(Vector(self.hw,  -self.hl, self.hh)),
+            self.relativePosition(Vector(self.hw,  self.hl,  -self.hh)),
+            self.relativePosition(Vector(-self.hw, self.hl,  -self.hh)),
+            self.relativePosition(Vector(-self.hw, -self.hl, -self.hh)),
+            self.relativePosition(Vector(self.hw,  -self.hl, -self.hh)),
+        )
+
+    @cached_property
+    def occupiedSpace(self):
+        """A region representing the space this object occupies"""
+        return MeshVolumeRegion(mesh=self.shape.mesh, \
+                dimensions=(self.width, self.length, self.height), \
+                position=self.position, rotation=self.orientation)
+
+    @cached_property
+    def boundingBox(self):
+        """A region representing this object's bounding box"""
+        return MeshVolumeRegion(self.occupiedSpace.mesh.bounding_box, center_mesh=False)
+
+    @cached_property
+    def inradius(self):
+        """A lower bound on the inradius of this object"""
+        # First check if all needed variables are defined. If so, we can
+        # compute the inradius exactly.
+        if not needsSampling([self.width, self.length, self.height, self.shape]):
+            shapeRegion = MeshVolumeRegion(mesh=self.shape.mesh, \
+                dimensions=(self.width, self.length, self.height))
```

Comment:
Can you avoid the construction of this separate region by just using `self.occupiedSpace.inradius`?

Reply:
This avoids a dependency on orientation, so we can get an exact inradius in more cases.